Webpack errors don't generate failure exit code

-   Laravel Mix Version: 6.0.2 (`npm list --depth=0`)
-   Node Version (`node -v`): 14.15.3
-   Yarn Version (`npm -v`): 1.22.5
-   OS: Fedora 33 (but running node in `node:14` docker image)

### Description:
Not sure if related to #2680 - please close if it is. After running `yarn run prod` on my GitLab CI job, mix failed with a few errors, but the CI job was considered successful which is undesirable as the pipeline continued to the next stage without the compiled assets. I'm currently using the workaround in https://github.com/JeffreyWay/laravel-mix/issues/2680#issue-771488674, so not sure if that is the reason.

But I would expect any webpack errors to result in a failure exit code.

### Steps To Reproduce:
```js
// webpack.mix.js
const mix = require('laravel-mix');

mix.js('unknown.js', 'bar.js')
    .webpackConfig({
        stats: "normal",
    })
```
```js
// package.json
"scripts": {
    "dev": "yarn run development",
    "development": "mix",
    "watch": "mix watch",
    "watch-poll": "mix watch -- --watch-options-poll=1000",
    "hot": "mix watch --hot",
    "prod": "yarn run production",
    "production": "mix --production"
},
```
```
[philbates@localhost myapp]$ node yarn run prod
Creating myapp_node_run ... done
yarn run v1.22.5
$ yarn run production
$ mix --production
[webpack-cli] Compilation finished
assets by status 0 bytes [cached] 1 asset

ERROR in /bar
Module not found: Error: Can't resolve '/usr/src/app/someunknown.js' in '/usr/src/app'

webpack 5.11.0 compiled with 1 error in 203 ms
Done in 2.28s.
[philbates@localhost myapp]$ echo $?
0
```
The last line shows that it exited with `0` exit code even though the command failed.

--- a/bin/cli.js
+++ b/bin/cli.js
@@ -65,9 +65,13 @@ async function executeScript(cmd, opts, args = []) {
         return process.stdout.write(script);
     }
 
-    spawn(script, {
+    const child = spawn(script, {
         stdio: 'inherit',
         shell: true
+    });
+
+    child.on('exit', code => {
+        process.exitCode = code;
     });
 }
 

--- a/test/integration/cli.js
+++ b/test/integration/cli.js
@@ -13,3 +13,23 @@ test('It can run the CLI', async t => {
     t.is('', stderr);
     t.is(undefined, error);
 });
+
+test('Missing config files result in non-zero exit code', async t => {
+    try {
+        await mix(['--mix-config=webpack.mix.does-not-exist']);
+
+        t.fail('Mix process exited normally when it should not have.');
+    } catch (err) {
+        t.not(0, err.code);
+    }
+});
+
+test('Webpack errors result in non-zero exit code', async t => {
+    try {
+        await mix(['--mix-config=webpack.mix.error']);
+
+        t.fail('Mix process exited normally when it should not have.');
+    } catch (err) {
+        t.not(0, err.code);
+    }
+});

